.github/workflows/main.yml

```
@@ -0,0 +1,82 @@
+name: Testing
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  NODE_OPTIONS: '--max_old_space_size=4096'
+
+jobs:
+  lint:
+    name: Linting Check
+    uses: the-guild-org/shared-config/.github/workflows/lint.yml@main
+    with:
+      script: yarn ci:lint
+    secrets:
+      githubToken: ${{ secrets.GITHUB_TOKEN }}
+  prettier-check:
+    name: 🧹 Prettier Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Master
+        uses: actions/checkout@v3
+      - name: Setup env
+        uses: the-guild-org/shared-config/setup@main
+        with:
+          nodeVersion: 17
+      - name: Prettier Check
+        run: yarn prettier:check
+  esm:
+    name: Testing exports integrity
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup env
+        uses: the-guild-org/shared-config/setup@main
+        with:
+          nodeVersion: 17
+      - name: Build
+        run: yarn build
+        env:
+          CI: true
+      - name: Test ESM & CJS integrity
+        run: yarn bob check
+  test:
+    name: Unit Test on Node ${{matrix.node_version}} (${{matrix.os}})
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+      - prettier-check
+      - esm
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        node_version: [12, 14, 16, 18]
```

do we need tests for 12?
```suggestion
        node_version: [14, 16, 18]
```